server/Arcadia.Assistant.Calendar/SickLeave/SendEmailSickLeaveActor.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Arcadia.Assistant.Calendar.SickLeave.Events;
+using System;
+using Akka.Actor;
+using Akka.DI.Core;
+using MimeKit;
+
+namespace Arcadia.Assistant.Calendar.SickLeave
+{
+    using Configuration.Configuration;
+    using MailKit.Net.Smtp;
+    using MailKit.Security;
+
+    public class SendEmailSickLeaveActor : UntypedActor
+    {
+        private readonly IEmailSettings mailConfig;
+        private readonly ISmtpSettings smtpConfig;
+        public SendEmailSickLeaveActor(IEmailSettings mailConfig, ISmtpSettings smtpConfig)
+        {
+            this.mailConfig = mailConfig;
+            this.smtpConfig = smtpConfig;
+        }
+
+        public static Props GetProps() => Context.DI().Props<SendEmailSickLeaveActor>();
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case SickLeaveIsApproved ev:
+                    SendEmail(ev);
+                    break;
+            }
+        }
+
+        private void SendEmail(SickLeaveIsApproved Event)
+        {
+            using (var client = new SmtpClient())
+            {
+                var message = CreateMimeMessage(String.Format(mailConfig.Body, Event.UserId, Event.TimeStamp.ToString()));
+
+                client.Connect(smtpConfig.Host, smtpConfig.Port, SecureSocketOptions.StartTls);
```

Comment:
Just for your education.

All I/O methods usually have asynchronous version (that's because they delegate actual work to OS - make a request and wait for response, for instance). The reason is that it doesn't lock a thread, and it can process other requests in the meantime.

Usually, in .NET (or in JS/NodeJS world) you do it using async/await operator. Here, in Akka, it's a bit harder and requires usage of Task.ContinueWith method. Here, however, it's unnecessary, because this code executes very rarely and we can keep it as is.